RecoEgamma/EgammaTools/interface/MVAValueMapProducer.h

```diff
@@ -60,54 +72,30 @@ MVAValueMapProducer<ParticleType>::MVAValueMapProducer(const edm::ParameterSet&
   srcMiniAOD_ = mayConsume<edm::View<ParticleType> >(iConfig.getParameter<edm::InputTag>("srcMiniAOD"));
 
   // Loop over the list of MVA configurations passed here from python and
-  // construct all requested MVA esimtators.
-  const std::vector<edm::ParameterSet>& mvaEstimatorConfigs
-    = iConfig.getParameterSetVector("mvaConfigurations");
-  for( auto &imva : mvaEstimatorConfigs ){
-
-    std::unique_ptr<AnyMVAEstimatorRun2Base> thisEstimator;
-    thisEstimator.reset(NULL);
-    if( !imva.empty() ) {
-      const std::string& pName = imva.getParameter<std::string>("mvaName");
-      // The factory below constructs the MVA of the appropriate type based
-      // on the "mvaName" which is the name of the derived MVA class (plugin)
-      AnyMVAEstimatorRun2Base *estimator = AnyMVAEstimatorRun2Factory::get()->create(pName, imva);
-      // Declare all event content, such as ValueMaps produced upstream or other,
-      // original event data pieces, that is needed (if any is implemented in the specific
-      // MVA classes)
-      //edm::ConsumesCollector &cc = consumesCollector();
-      estimator->setConsumes( consumesCollector() );
-
-      thisEstimator.reset(estimator);
-      
-    } else 
-      throw cms::Exception(" MVA configuration not found: ")
-	<< " failed to find proper configuration for one of the MVAs in the main python script " << std::endl;
-
-    // The unique pointer control is passed to the vector in the line below.
-    // Don't use thisEstimator pointer beyond the next line.
-    mvaEstimators_.emplace_back( thisEstimator.release() );
-
+  // construct all requested MVA esimtators.  
+  const auto& all_mvas = mva_cache->allMVAs();
+  for( auto mvaItr = all_mvas.begin(); mvaItr != all_mvas.end(); ++mvaItr ) {
+    // set the consumes
+    mvaItr->second->setConsumes(consumesCollector());
     //
     // Compose and save the names of the value maps to be produced
     //
-    const auto& currentEstimator = mvaEstimators_.back();
-    std::string thisValueMapName = currentEstimator->getName() + "Values";
-    std::string thisCategoriesMapName = currentEstimator->getName() + "Categories";
+    const auto& currentEstimator = mvaItr->second;
```

Comment:
@slava77 That is OK. The necessity of that commit seems to be dependent on
clang versions. It is updated in the VID PR anyway.

On Mon, Aug 10, 2015 at 10:56 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoEgamma/EgammaTools/interface/MVAValueMapProducer.h
> <https://github.com/cms-sw/cmssw/pull/10556#discussion_r36683452>:
>
> >      //
> >      // Compose and save the names of the value maps to be produced
> >      //
> > -    const auto& currentEstimator = mvaEstimators_.back();
> > -    std::string thisValueMapName = currentEstimator->getName() + "Values";
> > -    std::string thisCategoriesMapName = currentEstimator->getName() + "Categories";
> > +    const auto& currentEstimator = mvaItr->second;
>
> an equivalent of this commit is missing here
> lgray@e87c12e
> <https://github.com/lgray/cmssw/commit/e87c12ec391d39e393e5346d97d56a32919d6e42>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/10556/files#r36683452>.
>